digdag-server/src/main/java/io/digdag/server/rs/SessionResource.java

```diff
@@ -64,17 +65,20 @@ public SessionResource(
     @Path("/api/sessions")
     public RestSessionCollection getSessions(
             @QueryParam("last_id") Long lastId,
-            @QueryParam("page_size") Integer pageSize)
+            @QueryParam("page_size") Integer pageSize,
+            @QueryParam("page_number") Integer pageNumber)
     {
         int validPageSize = QueryParamValidator.validatePageSize(Optional.fromNullable(pageSize), MAX_SESSIONS_PAGE_SIZE, DEFAULT_SESSIONS_PAGE_SIZE);
 
         return tm.begin(() -> {
             ProjectStore rs = rm.getProjectStore(getSiteId());
             SessionStore ss = sm.getSessionStore(getSiteId());
 
-            List<StoredSessionWithLastAttempt> sessions = ss.getSessions(validPageSize, Optional.fromNullable(lastId));
+            Integer sessionRecordsNumber = ss.getTotalSessionsCount(Optional.fromNullable(lastId));
 
-            return RestModels.sessionCollection(rs, sessions);
+            Integer page = Optional.fromNullable(pageNumber).or(1);
```

Comment:
The variable name _page_ is uncertain.
It may be better to define QueryParamValidator.validatePageNumber and change the name to validPageNumber.